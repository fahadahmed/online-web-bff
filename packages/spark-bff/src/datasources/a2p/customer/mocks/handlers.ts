import { getAdminOverviewMock } from './getAdminOverview';
import { getCustomerOverviewMock } from './getCustomerOverview';
import { getContentProvidersMock } from './getContentProviders';
import { getServiceRequestsMock } from './getServiceRequests';
import { getShortcodesMock } from './getShortcodes';
import { getShortcodeMock } from './getShortcode';
import { getUsageMock } from './getUsage';
import { addContentProviderMock } from './addContentProvider';
import { addServiceRequestMock } from './addServiceRequest';
import { modifyShortcodeMock } from './modifyShortcode';
import { modifyContentProviderMock } from './modifyContentProvider';
import { deleteContentProviderMock } from './deleteContentProvider';

const handlers = [
  addContentProviderMock,
  addServiceRequestMock,
  getAdminOverviewMock,
  getCustomerOverviewMock,
  getContentProvidersMock,
  getServiceRequestsMock,
  getShortcodesMock,
  getShortcodeMock,
  getUsageMock,
  modifyShortcodeMock,
  modifyContentProviderMock,
  deleteContentProviderMock,
];

export default handlers;
