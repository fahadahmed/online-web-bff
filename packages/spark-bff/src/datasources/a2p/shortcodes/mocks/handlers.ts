import { getAdminShortcodesMock } from './getShortcodes';
import { getAdminShortcodesOverviewMock } from './getShortcodesOverview';

const handlers = [getAdminShortcodesMock, getAdminShortcodesOverviewMock];

export default handlers;
