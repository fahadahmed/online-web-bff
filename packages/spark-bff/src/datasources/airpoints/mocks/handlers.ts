import { rest } from 'msw';
import airpointsResponseMock from './airpoints-response.mock.json';
import addAirpointsResponseMock from './add-airpoints-response.mock.json';
import deleteAirpointsResponseMock from './delete-airpoints-response.mock.json';
import airpointsInactiveResponseMock from './airpoints-inactive-response.mock.json';
import addAirpointsErrorResponseMock from './add-airpoints-error-response.mock.json';
import deleteAirpointsErrorResponseMock from './delete-airpoints-error-response.mock.json';

interface AddAirpointsMockBody {
  accountNumber?: string;
  airpointsNumber?: string;
  firstName?: string;
  lastName?: string;
  status?: string;
}

export const airpointsResponseMocks = rest.get<AddAirpointsMockBody>(
  `http://testhost/v1/services/airpoints/:accountNumber`,
  async ({ params: { accountNumber } }, res, ctx) => {
    if (accountNumber === '123456789') {
      /* Simulate error responses */
      return res(ctx.status(200), ctx.json(airpointsResponseMock));
    }
    return res(ctx.status(200), ctx.json(airpointsInactiveResponseMock));
  },
);

export const updateAirpointsResponseMocks = rest.post<AddAirpointsMockBody>(
  `http://testhost/v1/services/airpoints/:accountNumber`,
  async (
    { params: { accountNumber }, body: { airpointsNumber } },
    res,
    ctx,
  ) => {
    /* add airpoints */
    if (airpointsNumber) {
      if (airpointsNumber === '911') {
        /* Simulate error responses */
        return res(ctx.status(400), ctx.json(addAirpointsErrorResponseMock));
      }
      return res(ctx.status(200), ctx.json(addAirpointsResponseMock));
    }
    /* delete airpoints */
    if (accountNumber) {
      if (accountNumber === '911') {
        /* Simulate error responses */
        return res(ctx.status(200), ctx.json(deleteAirpointsErrorResponseMock));
      }
      return res(ctx.status(200), ctx.json(deleteAirpointsResponseMock));
    }
    return null;
  },
);

const handlers = [airpointsResponseMocks, updateAirpointsResponseMocks];

export default handlers;
