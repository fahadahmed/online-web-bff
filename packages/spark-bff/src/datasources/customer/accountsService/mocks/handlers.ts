import { rest } from 'msw';
import accountSummaryMock from './account-summary.mock.json';
import emptyAccountListMock from './empty-account-list.mock.json';
import singleAccountListMock from './single-account-list.mock.json';
import accountListMock from './account-list.mock.json';
import guestAccountListMock from './guest-account-list.mock.json';
import unauthorizedAccountListMock from './unauthorized-account-list.mock.json';

const baseUrl = 'http://testhost/v1/customer/account';

export const accountSummaryHandler = rest.get<unknown, { accountId: string }>(
  `${baseUrl}/:accountId/summary`,
  async (req, res, ctx) => {
    const { accountId } = req.params;
    if (accountId.match(/^\d+$/) !== null) {
      return res(ctx.status(200), ctx.json(accountSummaryMock));
    }
    /* If the accountNumber does not match the given pattern, [0-9]+
    for example, 713012977, then it returns failure response, "Bad Request" */
    return res(ctx.status(400), ctx.json({}));
  },
);

export const accountListHandler = rest.get(
  `${baseUrl}/me/list`,
  async (_, res, ctx) => {
    return res(ctx.status(200), ctx.json(accountListMock));
  },
);

export const singleAccountListHandler = rest.get(
  `${baseUrl}/me/list`,
  async (_, res, ctx) => {
    return res(ctx.status(200), ctx.json(singleAccountListMock));
  },
);

export const emptyAccountListHandler = rest.get(
  `${baseUrl}/me/list`,
  async (_, res, ctx) => {
    return res(ctx.status(200), ctx.json(emptyAccountListMock));
  },
);

export const unauthorizedAccountListHandler = rest.get(
  `${baseUrl}/me/list`,
  async (_, res, ctx) => {
    return res(ctx.status(401), ctx.json(unauthorizedAccountListMock));
  },
);

export const guestAccountListHandler = rest.get(
  `${baseUrl}/me/list`,
  async (_, res, ctx) => {
    return res(ctx.status(403), ctx.json(guestAccountListMock));
  },
);

const handlers = [accountSummaryHandler, accountListHandler];

export default handlers;
