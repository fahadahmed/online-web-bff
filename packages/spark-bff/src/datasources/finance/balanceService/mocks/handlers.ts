import { rest } from 'msw';
import accountBalanceDetails from './account-balance.mock.json';

const baseUrl = 'http://testhost/v1/finance/balance/';

export const accountBalanceMock = rest.get<unknown, { accountNumber: string }>(
  `${baseUrl}account/:accountNumber`,
  async ({ params: { accountNumber } }, res, ctx) => {
    if (accountNumber.match(/^\d+$/) !== null) {
      return res(ctx.status(200), ctx.json(accountBalanceDetails));
    }
    /* If the accountNumber does not match the given pattern, [0-9]+
    for example, 713012977, then it returns failure response, "Bad Request" */
    return res(ctx.status(400), ctx.json({}));
  },
);

const handlers = [accountBalanceMock];

export default handlers;
