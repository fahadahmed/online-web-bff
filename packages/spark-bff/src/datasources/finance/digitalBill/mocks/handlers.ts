import { rest } from 'msw';
import accountBillingSummary from './account-billing-summary.mock.json';
import accountTransactions from './account-transactions.mock.json';

const baseUrl = 'http://testhost/v2/finance';

export const accountBillingSummaryMock = rest.get(
  `${baseUrl}/:accountNumber/bills/summary`,
  async (_, res, ctx) => {
    return res(ctx.json(accountBillingSummary));
  },
);

export const accountTransactionsMock = rest.get(
  `${baseUrl}/:accountNumber/bills/transactions`,
  async (_, res, ctx) => {
    return res(ctx.json(accountTransactions));
  },
);

const handlers = [accountBillingSummaryMock, accountTransactionsMock];

export default handlers;
