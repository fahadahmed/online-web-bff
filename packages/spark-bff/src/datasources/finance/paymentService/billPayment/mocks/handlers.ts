import { rest } from 'msw';
import poliPayBillSuccess from './poliPayBill-success.mock.json';
import poliPayBillValidationFailure from './poliPayBill-failure.mock.json';
import creditCardPayBillSuccess from './creditCard-payBill-success.mock.json';
import creditCardPayBillValidationFailure from './creditCard-payBill-failure.mock.json';
import deleteAutoPayBillSuccess from './delete-autoPayBill-success.mock.json';
import deleteAutoPayBillFailure from './delete-autoPayBill-failure.mock.json';

const baseUrl = 'http://testhost/v1/finance/payment';

export const addPoliPayBillMock = rest.post<unknown, { accountNumber: string }>(
  `${baseUrl}/polipay/bill/:accountNumber`,
  async ({ params: { accountNumber } }, res, ctx) => {
    /* If the accountNumber does not match the given pattern, [0-9]+
    for example, 789046383,then it returns failure response, "Request validation failure."  */
    if ((accountNumber as string).match(/^\d+$/) === null) {
      return res(ctx.status(400), ctx.json(poliPayBillValidationFailure));
    }
    return res(ctx.status(200), ctx.json(poliPayBillSuccess));
  },
);

export const addCreditCardPayBillMock = rest.post<
  unknown,
  { accountNumber: string }
>(
  `${baseUrl}/creditcard/bill/:accountNumber`,
  async ({ params: { accountNumber } }, res, ctx) => {
    /* If the accountNumber does not match the given pattern, [0-9]+
    for example, 789046383,then it returns failure response, "Request validation failure."  */
    if (accountNumber.match(/^\d+$/) === null) {
      return res(ctx.status(400), ctx.json(creditCardPayBillValidationFailure));
    }
    return res(ctx.status(200), ctx.json(creditCardPayBillSuccess));
  },
);

export const deleteAutoPayBillMock = rest.delete<
  unknown,
  { accountNumber: string; paymentMethodId: string }
>(
  `${baseUrl}/bill/:accountNumber/paymentsetup/:paymentMethodId`,
  async ({ params: { accountNumber, paymentMethodId } }, res, ctx) => {
    /** If accountNumber does not match the given pattern,
     for example, 746738998.
     Or If paymentMethodId does not match the given pattern,
     for example,"1-S0123-23"
     it returns failure response, "Request Validation Failure".
     */
    if (
      accountNumber.match(/^\d+$/) === null ||
      paymentMethodId.match(/^[\dA-Z][\dA-Z-]+[\dA-Z]$/) === null
    ) {
      return res(ctx.status(400), ctx.json(deleteAutoPayBillFailure));
    }
    return res(ctx.status(200), ctx.json(deleteAutoPayBillSuccess));
  },
);

const handlers = [
  addCreditCardPayBillMock,
  addPoliPayBillMock,
  deleteAutoPayBillMock,
];

export default handlers;
