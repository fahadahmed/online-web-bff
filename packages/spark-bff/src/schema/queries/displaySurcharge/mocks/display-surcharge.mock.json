{
  "PAY_BILL": {
    "operationType": "PAY_BILL",
    "masterCard": "3%",
    "visa": "3%",
    "amex": "2%",
    "dinersClub": "2%",
    "displayText": [
      "American Express and Diners Club incur a 2% surcharge.",
      "Visa/MasterCard, Prezzy Card, international credit, debit or charge cards incur a 3% surcharge."
    ]
  },
  "SETUP_MONTHLY_PAYMENT": {
    "operationType": "SETUP_MONTHLY_PAYMENT",
    "masterCard": "3%",
    "visa": "3%",
    "amex": "2%",
    "dinersClub": "2%",
    "displayText": [
      "Automated payments using a credit/debit card incurs a 1.5% surcharge (plus GST, if any). Surcharge applies."
    ]
  }
}
