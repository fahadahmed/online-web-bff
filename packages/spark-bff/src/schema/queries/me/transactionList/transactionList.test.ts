import { getTransactionListMock } from 'datasources/line/transactionsService/mocks/handlers';
import { createTestClient, gql } from 'test/apolloTestUtils';
import { server } from 'test/server';

beforeEach(() => {
  server.use(getTransactionListMock);
});

test('returns lineNumber transactions list', async () => {
  const { query } = createTestClient();
  const res = await query({
    query: gql`
      query {
        me {
          transactionList(
            lineNumber: "021023456"
            startDate: "01/01/21"
            endDate: "06/06/21"
            size: 6
          ) {
            value
            type
            startDateTime
            isDebit
            description
          }
        }
      }
    `,
  });

  expect(res.errors).toBeUndefined();
  expect(res.data!.me.transactionList).toMatchSnapshot();
});

test('returns lineNumber transactions list with paymentstatus APPLIED only', async () => {
  const { query } = createTestClient();
  const res = await query({
    query: gql`
      query {
        me {
          transactionList(
            lineNumber: "021023456"
            startDate: "01/01/21"
            endDate: "06/06/21"
            size: 6
            paymentStatus: APPLIED
          ) {
            value
            type
            startDateTime
            isDebit
            description
          }
        }
      }
    `,
  });

  expect(res.errors).toBeUndefined();
  expect(res.data!.me.transactionList).toMatchSnapshot();
});
