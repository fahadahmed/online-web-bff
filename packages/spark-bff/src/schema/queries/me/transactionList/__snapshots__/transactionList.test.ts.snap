// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`returns lineNumber transactions list 1`] = `
Array [
  Object {
    "description": "$30.00 POLi payment",
    "isDebit": false,
    "startDateTime": "2020-02-01T23:32:31.000+13:00",
    "type": "Payment",
    "value": 50,
  },
  Object {
    "description": "$30.00 POLi payment",
    "isDebit": true,
    "startDateTime": "2020-02-02T23:32:31.000+13:00",
    "type": "Payment",
    "value": 30,
  },
  Object {
    "description": "$30.00 POLi payment",
    "isDebit": false,
    "startDateTime": "2020-02-03T23:32:31.000+13:00",
    "type": "Payment",
    "value": 30,
  },
  Object {
    "description": "$90.00 POLi payment",
    "isDebit": false,
    "startDateTime": "2020-07-04T23:32:31.000+13:00",
    "type": "Payment",
    "value": 90,
  },
  Object {
    "description": "$70.00 POLi payment",
    "isDebit": true,
    "startDateTime": "2020-03-05T23:32:31.000+13:00",
    "type": "Payment",
    "value": 30,
  },
  Object {
    "description": "$70.00 POLi payment",
    "isDebit": true,
    "startDateTime": "2020-09-06T23:32:31.000+13:00",
    "type": "Payment",
    "value": 30,
  },
  Object {
    "description": "$15.00 Poli payment",
    "isDebit": true,
    "startDateTime": "2020-03-07T23:32:31.000+13:00",
    "type": "Payment",
    "value": 30,
  },
  Object {
    "description": "$70.00 POLi payment",
    "isDebit": false,
    "startDateTime": "2020-12-08T23:32:31.000+13:00",
    "type": "Payment",
    "value": 30,
  },
  Object {
    "description": "$50.00 POLi payment",
    "isDebit": true,
    "startDateTime": "2020-09-09T23:32:31.000+13:00",
    "type": "Payment",
    "value": 30,
  },
  Object {
    "description": "$70.00 POLi payment",
    "isDebit": true,
    "startDateTime": "2020-09-10T23:32:31.000+13:00",
    "type": "Payment",
    "value": 30,
  },
  Object {
    "description": "$100.00 ANZ payment",
    "isDebit": false,
    "startDateTime": "2020-09-11T23:32:31.000+13:00",
    "type": "Payment",
    "value": 30,
  },
  Object {
    "description": "$70.00 POLi payment",
    "isDebit": false,
    "startDateTime": "2020-09-12T23:32:31.000+13:00",
    "type": "Payment",
    "value": 30,
  },
  Object {
    "description": "$70.00 POLi payment",
    "isDebit": false,
    "startDateTime": "2020-09-13T23:32:31.000+13:00",
    "type": "Payment",
    "value": 30,
  },
  Object {
    "description": "$70.00 POLi payment",
    "isDebit": false,
    "startDateTime": "2020-09-14T23:32:31.000+13:00",
    "type": "Payment",
    "value": 30,
  },
  Object {
    "description": "$70.00 POLi payment",
    "isDebit": false,
    "startDateTime": "2020-09-15T23:32:31.000+13:00",
    "type": "Payment",
    "value": 30,
  },
  Object {
    "description": "$70.00 POLi payment",
    "isDebit": false,
    "startDateTime": "2020-09-16T23:32:31.000+13:00",
    "type": "Payment",
    "value": 30,
  },
  Object {
    "description": "$70.00 POLi payment",
    "isDebit": false,
    "startDateTime": "2020-09-17T23:32:31.000+13:00",
    "type": "Payment",
    "value": 30,
  },
]
`;

exports[`returns lineNumber transactions list with paymentstatus APPLIED only 1`] = `
Array [
  Object {
    "description": "$30.00 POLi payment",
    "isDebit": false,
    "startDateTime": "2020-02-01T23:32:31.000+13:00",
    "type": "Payment",
    "value": 50,
  },
  Object {
    "description": "$30.00 POLi payment",
    "isDebit": false,
    "startDateTime": "2020-02-03T23:32:31.000+13:00",
    "type": "Payment",
    "value": 30,
  },
  Object {
    "description": "$90.00 POLi payment",
    "isDebit": false,
    "startDateTime": "2020-07-04T23:32:31.000+13:00",
    "type": "Payment",
    "value": 90,
  },
  Object {
    "description": "$70.00 POLi payment",
    "isDebit": true,
    "startDateTime": "2020-03-05T23:32:31.000+13:00",
    "type": "Payment",
    "value": 30,
  },
  Object {
    "description": "$70.00 POLi payment",
    "isDebit": true,
    "startDateTime": "2020-09-06T23:32:31.000+13:00",
    "type": "Payment",
    "value": 30,
  },
  Object {
    "description": "$15.00 Poli payment",
    "isDebit": true,
    "startDateTime": "2020-03-07T23:32:31.000+13:00",
    "type": "Payment",
    "value": 30,
  },
  Object {
    "description": "$70.00 POLi payment",
    "isDebit": false,
    "startDateTime": "2020-12-08T23:32:31.000+13:00",
    "type": "Payment",
    "value": 30,
  },
  Object {
    "description": "$50.00 POLi payment",
    "isDebit": true,
    "startDateTime": "2020-09-09T23:32:31.000+13:00",
    "type": "Payment",
    "value": 30,
  },
  Object {
    "description": "$70.00 POLi payment",
    "isDebit": true,
    "startDateTime": "2020-09-10T23:32:31.000+13:00",
    "type": "Payment",
    "value": 30,
  },
  Object {
    "description": "$100.00 ANZ payment",
    "isDebit": false,
    "startDateTime": "2020-09-11T23:32:31.000+13:00",
    "type": "Payment",
    "value": 30,
  },
  Object {
    "description": "$70.00 POLi payment",
    "isDebit": false,
    "startDateTime": "2020-09-12T23:32:31.000+13:00",
    "type": "Payment",
    "value": 30,
  },
  Object {
    "description": "$70.00 POLi payment",
    "isDebit": false,
    "startDateTime": "2020-09-13T23:32:31.000+13:00",
    "type": "Payment",
    "value": 30,
  },
  Object {
    "description": "$70.00 POLi payment",
    "isDebit": false,
    "startDateTime": "2020-09-14T23:32:31.000+13:00",
    "type": "Payment",
    "value": 30,
  },
  Object {
    "description": "$70.00 POLi payment",
    "isDebit": false,
    "startDateTime": "2020-09-15T23:32:31.000+13:00",
    "type": "Payment",
    "value": 30,
  },
  Object {
    "description": "$70.00 POLi payment",
    "isDebit": false,
    "startDateTime": "2020-09-16T23:32:31.000+13:00",
    "type": "Payment",
    "value": 30,
  },
  Object {
    "description": "$70.00 POLi payment",
    "isDebit": false,
    "startDateTime": "2020-09-17T23:32:31.000+13:00",
    "type": "Payment",
    "value": 30,
  },
]
`;
